overlord/snapstate/handlers_components.go

```diff
@@ -67,6 +67,37 @@ func TaskComponentSetup(t *state.Task) (*ComponentSetup, *SnapSetup, error) {
 	return &compSetup, snapsup, nil
 }
 
+func TaskComponentSetups(t *state.Task) ([]*ComponentSetup, *SnapSetup, error) {
```

Comment:
Defining this function seems also premature, I worry that we define it but then forget about it and is left around.